lib/document/stateTransition/validation/data/fetchDocumentsFactory.js

```diff
@@ -5,44 +5,43 @@
 function fetchDocumentsFactory(stateRepository) {
   /**
    * @typedef fetchDocuments
-   * @param {string} dataContractId
    * @param {DocumentCreateTransition[]
    *        |DocumentReplaceTransition[]
    *        |DocumentDeleteTransition[]} documentTransitions
    * @return {Document[]}
    */
-  async function fetchDocuments(dataContractId, documentTransitions) {
-    // Group Document IDs by types
-    const documentIdsByTypes = documentTransitions.reduce((obj, document) => {
-      if (!obj[document.getType()]) {
+  async function fetchDocuments(documentTransitions) {
+    // Group document transitions by contracts and types
+    const transitionsByContractsAndTypes = documentTransitions.reduce((obj, dt) => {
+      const uniqueKey = `${dt.getDataContractId()}${dt.getType()}`;
```

Comment:
Maybe it's better to have a method for this?

Reply:
You mean like `document.getUniqueKey()`?

Reply:
Not sure it makes sense since we need it only in one place - here.

Reply:
@jawid-h  what do you think?

Reply:
Just a local identifier, so I think it is fine